internal/cluster/monitor.go

```diff
@@ -0,0 +1,71 @@
+package cluster
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/go-openapi/swag"
+	"github.com/openshift/assisted-service/pkg/requestid"
+
+	"github.com/jinzhu/gorm"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/models"
+	"github.com/sirupsen/logrus"
+)
+
+func (m *Manager) ClusterMonitoring() {
+	var (
+		clusters  []*common.Cluster
+		requestID = requestid.NewID()
+		ctx       = requestid.ToContext(context.Background(), requestID)
+		log       = requestid.RequestIDLogger(m.log, requestID)
+		err       error
+	)
+	if err = m.db.Preload("Hosts").Find(&clusters).Error; err != nil {
+		log.WithError(err).Errorf("failed to get clusters")
+		return
+	}
+	for _, cluster := range clusters {
+		if err = m.UpdateHostsAndClusterStatus(ctx, cluster, m.db, log); err != nil {
+			log.WithError(err).Errorf("failed to refresh cluster %s state", cluster.ID)
+			continue
+		}
+	}
+}
+
+func (m *Manager) UpdateHostsAndClusterStatus(ctx context.Context, cluster *common.Cluster, db *gorm.DB, log logrus.FieldLogger) error {
+	err := m.refreshClusterHosts(ctx, cluster, db, log)
+	if err != nil {
```

Comment:
if we are failing to update something in host? Don't we need still to try to refresh status of cluster? another way we can be stuck twice. in host status and in cluster status too.

Reply:
+1